distributions/openhab/src/main/resources/bin/update.ps1

```diff
@@ -38,83 +65,107 @@ Function Update-openHAB {
-            #$LegacyAddonsDownloadLocation="https://openhab.ci.cloudbees.com/job/openHAB-Distribution/lastSuccessfulBuild/artifact/distributions/openhab-addons-legacy/target/openhab-addons-legacy-$OHVersion.kar"
-            Write-Host -ForegroundColor Cyan "Downloading the openHAB $OHVersion distribution (snapshot)..."
-            Invoke-WebRequest -Uri $DownloadLocation -OutFile "openhab-$OHVersion.zip"
+            $AddonsDownloadLocation="https://openhab.ci.cloudbees.com/job/openHAB-Distribution/lastSuccessfulBuild/artifact/distributions/openhab-addons/target/openhab-addons-$OHVersion.kar"
+            $LegacyAddonsDownloadLocation="https://openhab.ci.cloudbees.com/job/openHAB-Distribution/lastSuccessfulBuild/artifact/distributions/openhab-addons-legacy/target/openhab-addons-legacy-$OHVersion.kar"
+            Write-Host -ForegroundColor Cyan "Downloading the openHAB $OHVersion distribution..."
+            DownloadFiles $DownloadLocation "$TempDir\openhab-$OHVersion.zip"
         } else {
             $DownloadLocation="https://bintray.com/openhab/mvn/download_file?file_path=org%2Fopenhab%2Fdistro%2Fopenhab%2F$OHVersion%2Fopenhab-$OHVersion.zip"
-            #$AddonsDownloadLocation="https://bintray.com/openhab/mvn/download_file?file_path=org%2Fopenhab%2Fdistro%2Fopenhab-addons%2F$OHVersion%2Fopenhab-addons-$OHVersion.kar"
-            #$LegacyAddonsDownloadLocation="https://bintray.com/openhab/mvn/download_file?file_path=org%2Fopenhab%2Fdistro%2Fopenhab-addons-legacy%2F$OHVersion%2Fopenhab-addons-legacy-$OHVersion.kar"
-            Write-Host -ForegroundColor Cyan "Downloading the openHAB $OHVersion distribution (release)..."
-            Start-BitsTransfer $DownloadLocation -Destination "openhab-$OHVersion.zip"
-        }
-
-        if (!(Test-Path "openhab-$OHVersion.zip")) {
-            throw "Couldn't download the archive, aborting"
+            $AddonsDownloadLocation="https://bintray.com/openhab/mvn/download_file?file_path=org%2Fopenhab%2Fdistro%2Fopenhab-addons%2F$OHVersion%2Fopenhab-addons-$OHVersion.kar"
+            $LegacyAddonsDownloadLocation="https://bintray.com/openhab/mvn/download_file?file_path=org%2Fopenhab%2Fdistro%2Fopenhab-addons-legacy%2F$OHVersion%2Fopenhab-addons-legacy-$OHVersion.kar"
+            Write-Host -ForegroundColor Cyan "Downloading the openHAB $OHVersion distribution..."
+            DownloadFiles $DownloadLocation "$TempDir\openhab-$OHVersion.zip"
         }
 
         Write-Host -ForegroundColor Cyan "Making a backup in $OHDirectory\backup..."
-        mkdir backup | Out-Null
-        Copy-Item $OHDirectory\conf $OHDirectory\backup\conf -Recurse -Force
-        Copy-Item $OHDirectory\runtime $OHDirectory\backup\runtime -Recurse -Force
-        Copy-Item $OHDirectory\userdata $OHDirectory\backup\userdata -Recurse -Force
+        if (Test-Path "$OHDirectory\backup")
```

Comment:
Is this temporary? If not, should it use the backup script you've already implemented?

Reply:
Completely reworked this to use new backup.ps1 file